recipes/libcap/all/test_package/conanfile.py

```diff
@@ -1,18 +1,38 @@
 import os
 
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake
+from conan.tools.env import Environment
+from conan.tools.cmake import cmake_layout
+
+required_conan_version = ">=1.38.0"
 
 
 class LibcapTestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "CMakeToolchain", "PkgConfigDeps", "VirtualBuildEnv", \
+        "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        env = Environment()
+        env.prepend_path("PKG_CONFIG_PATH", self.generators_folder)
```

Comment:
For some reason, `pkg_check_modules` tries to find `.pc` files not in the `generator_folder`, but in the `generator_folder/lib/pkgconfig`. I thought the `PkgConfigDeps` generator should resolve that (via env vars maybe), but it doesn't. Not sure if I am doing something wrong or if it is a bug. I didn't find any examples of using `CMakeToolchain` and `PkgConfigDeps` generators together.

Reply:
:wave: @franramirez688 @czoido is this the right way to use CMakeToolchain and PkgConfigDeps generators together?

Reply:
> For some reason, pkg_check_modules tries to find .pc files not in the generator_folder, but in the generator_folder/lib/pkgconfig.

After digging deeper into it, I figured out that CMake is adding the suffix `lib/pkgconfig` to the `CMAKE_PREFIX_PATH` (pointing to the `package_folder` where the `*.pc` files are created). You can have a look at the issue https://gitlab.kitware.com/cmake/cmake/-/issues/18150

> I thought the PkgConfigDeps generator should resolve that (via env vars maybe), but it doesn't

Yes, you're right, but it should not be necessary as far as CMake should use that `CMAKE_PREFIX_PATH` by default. As I said above, CMake is adding that suffix, so it's getting failed because there are no PC files there.

> is this the right way to use CMakeToolchain and PkgConfigDeps generators together?

Given that information, I tend to say that yes, it could be one way to solve that issue, and I think there could be another workaround like:

```python
    def build(self):
        pkg_lib_folder = os.path.join(self.generators_folder, "lib", "pkgconfig")
        copy(self, "*.pc", self.generators_folder, pkg_lib_folder)
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```
Anyway, we could discuss if it could be a chance to create a mechanism to add the `PKG_CONFIG_PATH` via CMake build helper or even CMakeToolchain to avoid those workarounds.